File: codewars/javascript/maze.js
function pathFinder(maze)
{
	N = ~~(Math.sqrt(maze.length))
	console.log(N)
	maze = maze.split("\n").join("")
	console.log(maze, maze.length)
	visited = {}
	let nbrs = [[0, 1], [0, -1], [1, 0], [-1, 0]]
 
	function dfs(x, y)
	{
		if (x == N-1 && y == N-1) {
			console.log(maze[x*N + y], x*N + y)
			if (maze[x * N + y] == '.')
				return true
			return false
		}
		visited[[x,y]] = true
		for (let i = 0; i < 4; i++) {
			let d = nbrs[i]
			let nx = x + d[0]
			let ny = y + d[1]
			if (nx < 0  || nx >= N || ny < 0 || ny >= N 
						|| maze[nx*N + ny] == 'w'
						|| visited[[nx, ny]])
				continue
			if ((r = dfs(nx, ny)))
				return r
		}
		return false
	}
	return dfs(0, 0, maze)
}

maze = (`......
......
......
......
.....w
....w.`);

console.log(pathFinder(maze))

File: codewars/python/maze.py
def has_exit(maze):
    c = 0
    for p, v in enumerate(maze):
        if 'k' in v:
            c += 1
            kpos = (p, v.index('k'))
    if c != 1:
        raise Exception()
    if (len(maze) == 1 and len(maze[0]) == 1):
        return True
    
    x, y = kpos
    dir = [(0, -1), (0, 1), (1, 0), (-1, 0)]
    visited = set()
    L = len(maze)
    W = len(maze[0])
    visited.add((x,y))
    
    def dfs(maze, x, y):
        print(x, y)
        if x == 0 or y == 0 or x == L-1 or y == W-1:
            if maze[x][y] == ' ': 
                return True
                
        for dx, dy in dir:
            fx, fy = (x + dx, y + dy)
            if fx < 0 or fy < 0 or fx >= L or fy >= W or \
                      (fx, fy) in visited or \
                      maze[fx][fy] == '#':
                continue
            visited.add((fx, fy))
            if dfs(maze, fx, fy):
                return True
        return False
